cmd/boulder-remoteva/main.go

```diff
@@ -0,0 +1,126 @@
+	vacfg "github.com/letsencrypt/boulder/va/config"
+	vapb "github.com/letsencrypt/boulder/va/proto"
+)
+
+type Config struct {
+	RVA struct {
+		vacfg.Shared
+	}
+
+	Syslog        cmd.SyslogConfig
+	OpenTelemetry cmd.OpenTelemetryConfig
+}
+
+func main() {
+	grpcAddr := flag.String("addr", "", "gRPC listen address override")
+	debugAddr := flag.String("debug-addr", "", "Debug server address override")
+	configFile := flag.String("config", "", "File path to the configuration file for this service")
+	flag.Parse()
+	if *configFile == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	var c Config
+	err := cmd.ReadConfigFile(*configFile, &c)
+	cmd.FailOnError(err, "Reading JSON config file into config structure")
+
+	features.Set(c.RVA.Features)
+
+	if *grpcAddr != "" {
+		c.RVA.GRPC.Address = *grpcAddr
+	}
+	if *debugAddr != "" {
+		c.RVA.DebugAddr = *debugAddr
+	}
+
+	scope, logger, oTelShutdown := cmd.StatsAndLogging(c.Syslog, c.OpenTelemetry, c.RVA.DebugAddr)
+	defer oTelShutdown(context.Background())
+	logger.Info(cmd.VersionString())
+
+	if c.RVA.DNSTimeout.Duration == 0 {
+		cmd.Fail("'dnsTimeout' is required")
+	}
+	dnsTries := c.RVA.DNSTries
+	if dnsTries < 1 {
+		dnsTries = 1
+	}
+	clk := cmd.Clock()
+
+	var servers bdns.ServerProvider
+	proto := "udp"
+	if features.Get().DOH {
+		proto = "tcp"
+	}
+
+	if len(c.RVA.DNSStaticResolvers) != 0 {
+		servers, err = bdns.NewStaticProvider(c.RVA.DNSStaticResolvers)
+		cmd.FailOnError(err, "Couldn't start static DNS server resolver")
+	} else {
+		servers, err = bdns.StartDynamicProvider(c.RVA.DNSProvider, 60*time.Second, proto)
+		cmd.FailOnError(err, "Couldn't start dynamic DNS server resolver")
+	}
+	defer servers.Stop()
+
+	tlsConfig, err := c.RVA.TLS.Load(scope)
+	cmd.FailOnError(err, "tlsConfig config")
+
+	var resolver bdns.Client
+	if !c.RVA.DNSAllowLoopbackAddresses {
+		resolver = bdns.New(
+			c.RVA.DNSTimeout.Duration,
+			servers,
+			scope,
+			clk,
+			dnsTries,
+			logger,
+			tlsConfig)
+	} else {
+		resolver = bdns.NewTest(
+			c.RVA.DNSTimeout.Duration,
+			servers,
+			scope,
+			clk,
+			dnsTries,
+			logger,
+			tlsConfig)
+	}
+
+	vai, err := va.NewValidationAuthorityImpl(
+		resolver,
+		nil, // A remote VA itself will not have remote VAs.
+		0,   // MaxRemoteValidationFailures is intended for the primary VA only
+		c.RVA.UserAgent,
+		c.RVA.IssuerDomain,
+		scope,
+		clk,
+		logger,
+		c.RVA.AccountURIPrefixes)
+	cmd.FailOnError(err, "Unable to create remote VA server")
+
+	start, err := bgrpc.NewServer(c.RVA.GRPC, logger).Add(
+		&vapb.VA_ServiceDesc, vai).Add(
+		&vapb.CAA_ServiceDesc, vai).Build(tlsConfig, scope, clk)
+	cmd.FailOnError(err, "Unable to setup remote VA gRPC server")
+
+	cmd.FailOnError(start(), "remote VA gRPC service failed")
+}
+
+func init() {
+	// TODO(#5294) After we have separate RVA and VA binaries deployed,
+	// re-register boulder-remoteva2 as boulder-remoteva.
+	cmd.RegisterCommand("boulder-remoteva2", main, &cmd.ConfigValidator{Config: &Config{}})
```

Comment:
Can we just call this command "remoteva"?  Then when it's invoked, it'll be "boulder remoteva" instead of "boulder boulder-remoteva2".

We've kept the boulder- prefix on existing commands to avoid any impact on logging, but I think we're going to have to have to change the name here.  In that case, we can drop the prefix, and thus also don't need to suffix on a "2" either.

Reply:
I've renamed it again to `boulder-remoteva` instead of `boulder-remoteva2` and `boulder-rva`. I don't want to strip `boulder-` because then I would feel the need to clean up the rest of the commands to conform.

Reply:
I think removing "boulder-" from commands going forward is the correct decision, but we should absolutely not change the existing commands as that would break several log filters/searches..

Reply:
I'm pretty sure `boulder-*` is reserved for long-running boulder components versus one-off tools like id-exporter, notify-mailer, expiry-mailer, observer, etc.

Reply:
It's definitely inconsistent today: 

`akamai-purger`, `bad-key-revoker`, `crl-storer`, `crl-updater`, `expiration-mailer`, `log-validator`, `nonce-service`, `ocsp-responder` are all long-lived services without a boulder prefix. That's more than the 7 boulder-* components (-ra, -ca, -va, -sa, -wfe2, -observer, -publisher).

I think the prefix made sense when there was a separate binary for each command, but we're using subcommands now, so I think the prefix has outlasted its utility.

Reply:
I've renamed the binary to just `remoteva`.